earth2grid/_regrid.py

```diff
@@ -12,12 +12,48 @@
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
 # limitations under the License.
+import math
+from typing import Dict, Sequence
+
 import einops
 import netCDF4 as nc
 import torch
+from scipy import spatial
+
+from earth2grid.spatial import ang2vec, haversine_distance
+
+
+class Regridder(torch.nn.Module):
+    """Regridder to n points, with p nonzero maps weights
+
+    Forward:
+        (*, m) -> (*,) + shape
+    """
+
+    def __init__(self, shape: Sequence[int], p: int):
+        super().__init__()
+        self.register_buffer("index", torch.empty(*shape, p, dtype=torch.long))
+        self.register_buffer("weight", torch.ones(*shape, p))
+
+    def forward(self, z):
+        *shape, x = z.shape
+        zrs = z.view(-1, x).T
+
+        *output_shape, p = self.index.shape
+        index = self.index.view(-1, p)
+        weight = self.weight.view(-1, p)
 
-from earth2grid import base, healpix
-from earth2grid.latlon import LatLonGrid
+        # using embedding bag is 2x faster on cpu and 4x on gpu.
+        output = torch.nn.functional.embedding_bag(index, zrs, per_sample_weights=weight, mode='sum')
+        output = output.T.view(*shape, -1)
+        return output.reshape(list(shape) + output_shape)
+
+    @staticmethod
```

Comment:
Shouldn't this be a `classmethod`?

Reply:
I don't intend to inherit this object so used staticmethod and explicitly refer to the class name instead:
```
        n, p = d["index"].shape
        regridder = Regridder((n,), p)
        regridder.load_state_dict(d)
```